athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/DateTimeFormatterUtil.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Transforms big decimal value to LocalDate using the provided default format
+     *
+     * @param value raw value to be transformed to LocalDate
+     * @param defaultTimeZone default timezone to be applied
+     * @return LocalDate object parsed from value
+     */
+    public static LocalDate bigDecimalToLocalDate(BigDecimal value, ZoneId defaultTimeZone)
+    {
+        Date dateTransformedValue = bigDecimalToDate(value);
+        if (dateTransformedValue == null) {
+            return null;
+        }
+        return dateTransformedValue.toInstant()
+                .atZone(defaultTimeZone)
+                .toLocalDate();
+    }
+
+    /**
+     * Transforms big decimal value to LocalDateTime using the provided default format
+     *
+     * @param value raw value to be transformed to LocalDateTime
+     * @param defaultTimeZone default timezone to be applied
+     * @return LocalDateTime object parsed from value
+     */
+    public static LocalDateTime bigDecimalToLocalDateTime(BigDecimal value, ZoneId defaultTimeZone)
+    {
+        Date dateTransformedValue = bigDecimalToDate(value);
+        if (dateTransformedValue == null) {
+            return null;
+        }
+        return dateTransformedValue.toInstant()
+                .atZone(defaultTimeZone)
+                .toLocalDateTime();
+    }
+
+    /**
+     * Infers the date format to be used for the values in that column by passing through an array of supported formats
+     * this is then cached in DDDBRecordMetadata and reused so that the values in the same column
+     * does not have to be re-evaluated/parsed
+     *
+     * @param value string value to infer a date time format from
+     * @return string representing the datetime format
+     *         null if it does not match any of the supported formats
+     */
+    public static String inferDateTimeFormat(String value)
+    {
+        for (String datetimeFormat : SUPPORTED_DATETIME_FORMAT) {
+            try {
+                DateUtils.parseDate(value, datetimeFormat);
+                logger.info("Inferred format {} for value {}", datetimeFormat, value);
+                return datetimeFormat;
+            }
+            catch (ParseException ex) {
+                continue;
+            }
+        }
+        logger.error("Failed to infer format for {}", value);
+        return null;
+    }
+
+    /**
+     * Parses the given raw string to the provided date format
+     * @param value raw string value to be parsed to Date object
+     * @param dateFormat date format to be used to parse
+     * @return date object that is parsed from raw value
+     */
+    private static Date stringToDate(String value, String dateFormat)
+    {
+        try {
+            return DateUtils.parseDate(value, dateFormat);
+        }
+        catch (ParseException | IllegalArgumentException ex) {
+            logger.error("Encountered value {} that is not consistent with inferred date/datetime format {}" +
```

Comment:
warn since we fallback?

and is there any value in logging the exception message or stacktrace?

Reply:
I dont think so, i think the message shows exactly what the input was and what format it was trying to use.